src/main/java/com/petros/bring/context/AnnotationConfigApplicationContext.java

```diff
@@ -20,20 +19,15 @@ public AnnotationConfigApplicationContext(BeanDefinitionRegistry registry, List<
     }
 
     private void registerBean(BeanDefinition beanDefinition) {
-        validate(beanDefinition);
-        getBean(registry.getBeanTypeByName(beanDefinition.getName()));
+        if (needsRegistration(beanDefinition)) {
+            getBean(registry.getBeanTypeByName(beanDefinition.getName()));
```

Comment:
Good idea, but still it is possible that we have to create prototype objects during startup if singleton bean depends on it.